### PR TITLE
Add :flash target for mdloader

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Not sure how much value it adds... but why not add some consistency.

### TODO
- [ ] Add docs
    * Current limitations
      * `applet-flash-samd51j18a.bin` must be installed to the path alongside `mdloader`

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
@XScorpion2 @MatthewRobo want to take this code change for a spin?

**Comment by tzarc** [comment]
Fails on Windows unless I put `applet-flash-samd51j18a.bin` in `qmk_firmware` root.

**Comment by tzarc** [comment]
Have logged https://github.com/Massdrop/mdloader/issues/21.

**Comment by ash0x0** [comment]
Recipe in https://github.com/qmk/qmk_firmware/pull/8711/commits/158ef4d86e1c4372693dfcf719dd7199274059f7 works for me, the one in https://github.com/qmk/qmk_firmware/pull/8711/commits/e22153cc2bb503ce9c93fb33ca741ce26bda01fe doesn't, simple fix, I'll make a suggestion.
I do execute however from qmk_firmware root which also has both mdloader_linux and the applet. Can't attest to windows issues.
I think it's best to just assume that mdloader (windows, linux or mac, note they have different names for the executable) and the applet are both in qmk root and have that be that. There's no scenario in which the user just sets flash target and have it just work without placing the applet and the loader somewhere so might as well just have them both in qmk root and not require setting path and call it a day.

File: tmk_core/arm_atsam.mk [hunk 1]
```diff
@@ -55,5 +55,21 @@ bin: $(BUILD_DIR)/$(TARGET).hex
 	$(OBJCOPY) -Iihex -Obinary $(BUILD_DIR)/$(TARGET).hex $(BUILD_DIR)/$(TARGET).bin
 	$(COPY) $(BUILD_DIR)/$(TARGET).bin $(TARGET).bin;
 
+MDLOADER_CLI ?= mdloader
+define EXEC_MDLOADER
+	$(eval ABS_BUILD_FILE := $(shell realpath $(BUILD_DIR)/$(TARGET).bin))
+
+	# Assume user has installed applet-flash-*.bin to path to avoid Massdrop/mdloader/issues/21
+	cd $$(dirname $$(which $(MDLOADER_CLI))) && ./$(MDLOADER_CLI) --first --download $(ABS_BUILD_FILE) --restart
```
Comment (ash0x0) [inline tmk_core/arm_atsam.mk:63]:
```suggestion
	cd $$(dirname $$(which $(MDLOADER_CLI))) && $(MDLOADER_CLI) --first --download $(ABS_BUILD_FILE) --restart
```

**Comment by zvecr** [comment]
The assumption here is its installed to the path, not dropped into the root of qmk_firmware. Also mdloader only has different names if you use the prebuilt binary files (<https://github.com/Massdrop/mdloader/blob/master/Makefile#L1>).

**Comment by tzarc** [comment]
I'd really like to avoid dropping closed-source binaries into the root of the repo (ideally not even in the repo in the first place, but eh) -- ideally we could put it into `util/` or something and change where we `cd` to for flashing?

**Comment by zvecr** [comment]
I would rather implement this like bootloadhid, where we tackle the general availability issue as part of install, than put extra files in util. TBH i've poured way more effort into this than I care to, so i'm just going to close this.

**Comment by ash0x0** [comment]
Don't think you should close it. No harm in adding it and it makes things slightly easier for now. So long as the windows issue is resolved, it works fine as is. If you don't have time to work on it anymore, with your permission I can pick it up.

**Pull request closed without merging**